sdk/python/lib/pulumi/output.py

```diff
@@ -307,7 +307,16 @@ def from_input(val: Input[T_co]) -> "Output[T_co]":
 
         # Is a (non-empty) dict or list? Recurse into the values within them.
         if val and isinstance(val, dict):
-            o_dict: Output[dict] = Output.all(**val)
+            # The keys themselves might be outputs, so we can't just pass `**val` to all.
+            kvs = list(val.items())
+            keys = list(map(lambda kv: kv[0], kvs))
+            values = list(map(lambda kv: kv[1], kvs))
```

Comment:
minor: Might be just me, but a list comprehension would read better here:

```suggestion
            keys = [kv[0] for kv in kvs]
            values = [kv[1] for kv in kvs]
```

An alternative is to just use `keys()` and `values()` methods: they're [guaranteed to be in the same order](https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects).

```python
keys = list(vals.keys())
values = list(vals.values())
```

But I can see why converting ordered pairs to lists feels safer than converting two separate iterators.

Reply:
yeh list comprehension is fine, that's more python like. I wasn't sure if they would be the same order, but I can change to that with a comment saying so.